### PR TITLE
Release v3.0.0-preview2

### PR DESCRIPTION
**Pull request by wangyum**
Include `3.0.0-preview2` release in the download page and documentation.md, add news post to announce Spark 3.0.0 preview2 release.

### CODE REVIEW

File: site/developer-tools.html [hunk 1]
```diff
@@ -479,7 +479,7 @@ <h4>Solving a binary incompatibility</h4>
 <p>For the problem described above, we might add the following:</p>
 
 <figure class="highlight"><pre><code class="language-scala" data-lang="scala"><span class="c1">// [SPARK-zz][CORE] Fix an issue
-</span><span class="nv">ProblemFilters</span><span class="o">.</span><span class="py">exclude</span><span class="o">[</span><span class="kt">DirectMissingMethodProblem</span><span class="o">](</span><span class="s">"org.apache.spark.SomeClass.this"</span><span class="o">)</span></code></pre></figure>
+</span><span class="nc">ProblemFilters</span><span class="o">.</span><span class="n">exclude</span><span class="o">[</span><span class="kt">DirectMissingMethodProblem</span><span class="o">](</span><span class="s">"org.apache.spark.SomeClass.this"</span><span class="o">)</span></code></pre></figure>
```
Comment (srowen) [inline site/developer-tools.html:482]: Just checking, are you using the latest jekyll? I know this caused variations on classes like this. I'm not sure off the top of my head whether this is updating or reverting what the latest one does.

Comment (wangyum) [reply site/developer-tools.html:482]: It is generated by `v3.8.5`. I updated jekyll to `v4.0.0`: https://github.com/apache/spark-website/pull/245/commits/27dac9e39da700fa22b8e461b7ea21f3249cf327

**Review by srowen — COMMENTED** [review]
Content looks OK. OK to merge if this is the result of the latest jekyll, rouge, etc

**Review by HyukjinKwon — APPROVED** [review]

**Review by maropu — APPROVED** [review]

File: news/_posts/2019-12-23-spark-3.0.0-preview2.md [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+- News
+tags: []
+status: publish
+type: post
+published: true
+meta:
+  _edit_last: '4'
+  _wpas_done_all: '1'
+---
+To enable wide-scale community testing of the upcoming Spark 3.0 release, the Apache Spark community has posted a <a href="https://archive.apache.org/dist/spark/spark-3.0.0-preview2/">Spark 3.0.0 preview2 released</a>. This preview is <b>not a stable release in terms of either API or functionality</b>, but it is meant to give the community early access to try the code that will become Spark 3.0. If you would like to test the release, please download it, and send feedback using either the <a href="https://spark.apache.org/community.html">mailing lists</a> or <a href="https://issues.apache.org/jira/browse/SPARK/?selectedTab=com.atlassian.jira.jira-projects-plugin:summary-panel">JIRA</a>.
```
Comment (gatorsmile) [inline news/_posts/2019-12-23-spark-3.0.0-preview2.md:14]: Could we add one more sentence? The doc is not easy for end users to find. 

> The documentation is available at the [link](https://spark.apache.org/docs/3.0.0-preview2/).

Comment (gatorsmile) [reply news/_posts/2019-12-23-spark-3.0.0-preview2.md:14]: Spark 3.0.0 preview2 released
->
Spark 3.0.0 preview2 release

File: news/_posts/2019-12-23-spark-3.0.0-preview2.md [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+
+The Spark issue tracker already contains a list of <a href="https://issues.apache.org/jira/browse/SPARK-26078?jql=statusCategory%20%3D%20done%20AND%20project%20%3D%2012315420%20AND%20fixVersion%20%3D%2012339177%20ORDER%20BY%20priority%20DESC%2C%20key%20ASC">features in 3.0</a>.
```
Comment (gatorsmile) [inline news/_posts/2019-12-23-spark-3.0.0-preview2.md:16]: Can we change the link to a more printable list?

https://issues.apache.org/jira/sr/jira.issueviews:searchrequest-printable/temp/SearchRequest.html?jqlQuery=statusCategory+%3D+done+AND+project+%3D+12315420+AND+fixVersion+%3D+12339177+ORDER+BY+priority+DESC%2C+key+ASC&tempMax=1000

**Pull request closed without merging**